src/Validator/Rules/DisableIntrospection.php

```diff
@@ -52,6 +52,6 @@ public static function introspectionDisabledMessage()
 
     protected function isEnabled()
     {
-        return $this->isEnabled !== self::DISABLED;
+        return $this->isEnabled !== static::DISABLED;
```

Comment:
@simPod can we turn off this sniffer?

Reply:
Hard to say and evaluate for me as I use `final` everywhere so this is not my daily bread to deal with.

But personally I'm against this as constants are not meant to be overridden and abstract static methods should be used instead. Not sure what LSB for constants implies in the future.

Reply:
I generally agree with that sentiment and am moving towards doing the same in projects that are self-contained. It is always possible to change the code there. Consumers of this library have no choice but to copy-paste code if we close a class for extension.

The late static binding might be a footgun here, but I feel like it only adds to the usefulness of this library and does not take away from it.

Reply:
I could easily change this back to `self`, because I agree that constants shouldn't be overridden. Someone could then still extend/override the `isEnabled` method without changing the constant.

Reply:
> Consumers of this library have no choice but to copy-paste code if we close a class for extension.

That would exactly be my suggestion :D to copy-paste it and do whatever you want with it, you're on your own.

My way is fully typed, fully strict, etc. blabla code so I can't encourage doing this. Don't consider me a blocker though, I'd stayed silent if you didn't mention me 🙃